cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ApplicationLifecycleServiceTest.java

```diff
@@ -104,10 +104,22 @@ public void stop(ProgramId programId) throws Exception {
 
   @Test
   public void testAppDeletionMultipleNS() throws Exception {
+    artifactRepository.addArtifact(
+      NamespaceId.DEFAULT.artifact(WORDCOUNT_APP_NAME, "1.0.0-SNAPSHOT").toId(), wordCountAppJar);
+    artifactRepository.addArtifact(
+      TEST_NAMESPACE_META1.getNamespaceId().artifact(WORDCOUNT_APP_NAME, "1.0.0-SNAPSHOT").toId(), wordCountAppJar);
 
     // deploy same app in two namespaces
-    deploy(WordCountApp.class, Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE1);
```

Comment:
hum.. maybe. Currently the deploy method is doing something very funny https://github.com/caskdata/cdap/blob/develop/cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java#L493